tests/cpp/test_instructions.cpp

```diff
@@ -0,0 +1,52 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <gtest/gtest.h>
+
+#include <fusion.h>
+#include <fusion_guard.h>
+#include <ops/all_ops.h>
+#include <runtime/executor.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+
+namespace nvfuser {
+
+using InstructionsTest = NVFuserTest;
```

Comment:
Just a nitpick, but I was a bit puzzled with the naming or categorization of tests. Would there be any other tests you'd anticipate to be categorized here? If not, I'd say just `NVFuserTest` in test_gpu3.cpp would be fine.

Reply:
"Instruction" is a very generic name. I'd suggest something like `test_math_opt.cpp`, `using MathOptTest = NvFuserTest`.